homeassistant/components/mastodon/entity.py

```diff
@@ -0,0 +1,47 @@
+"""Base class for Mastodon entities."""
+
+from homeassistant.helpers.device_registry import DeviceEntryType, DeviceInfo
+from homeassistant.helpers.entity import EntityDescription
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DEFAULT_NAME, DOMAIN, INSTANCE_VERSION
+from .coordinator import MastodonConfigEntry, MastodonCoordinator
+from .utils import construct_mastodon_username
+
+
+class MastodonEntity(CoordinatorEntity[MastodonCoordinator]):
+    """Defines a base Mastodon entity."""
+
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        coordinator: MastodonCoordinator,
+        description: EntityDescription,
+        data: MastodonConfigEntry,
+    ) -> None:
+        """Initialize Mastodon entity."""
+        super().__init__(coordinator)
+        unique_id = data.unique_id
+        assert unique_id is not None
+        self._attr_unique_id = f"{unique_id}_{description.key}"
+
+        # Legacy yaml config default title is Mastodon, don't make name Mastodon Mastodon
+        name = "Mastodon"
+        if data.title != DEFAULT_NAME:
+            name = f"Mastodon {data.title}"
+
+        full_account_name = construct_mastodon_username(
+            data.runtime_data.instance, data.runtime_data.account
+        )
+
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, unique_id)},
+            manufacturer="Mastodon gGmbH",
```

Comment:
That looks like a `g` too much

```suggestion
            manufacturer="Mastodon GmbH",
```

Reply:
That's what it has at the bottom of the about page here https://joinmastodon.org
The extra g signifies a non-profit company under German law according to a quick Google.

Reply:
Ooh, never heard of that, but that makes sense